protocols/kad/src/handler.rs

```diff
@@ -760,16 +760,6 @@ impl ConnectionHandler for Handler {
             });
         }
 
-        let no_streams = self.outbound_substreams.is_empty() && self.inbound_substreams.is_empty();
-        self.keep_alive = match (no_streams, self.keep_alive) {
-            // No open streams. Preserve the existing idle timeout.
-            (true, k @ KeepAlive::Until(_)) => k,
-            // No open streams. Set idle timeout.
-            (true, _) => KeepAlive::Until(Instant::now() + self.idle_timeout),
```

Comment:
Same as for gossipsub, we should deprecate `idle_timeout` first and then remove it with the next breaking release. That will reduce this function down to essentially just checking for existing streams which we no longer need to do with the changes in this PR!

Reply:
reverted e5a765f